cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/sampling/CruiseControlMetricsProcessor.java

```diff
@@ -137,29 +144,27 @@ void clear() {
    * @param cluster Kafka cluster
    * @param partitionsDotNotHandled The partitions to get samples. The topic partition name may have dots.
    * @param partitionMetricSamples The set to add the partition samples to.
-   * @return The number of skipped partitions.
+   * @return The number of skipped partitions by broker ids.
    */
-  private int addPartitionMetricSamples(Cluster cluster,
-                                        Set<TopicPartition> partitionsDotNotHandled,
-                                        Set<PartitionMetricSample> partitionMetricSamples) {
-    int skippedPartition = 0;
+  private Map<Integer, Integer> addPartitionMetricSamples(Cluster cluster,
+                                                          Set<TopicPartition> partitionsDotNotHandled,
+                                                          Set<PartitionMetricSample> partitionMetricSamples) {
+    Map<Integer, Integer> skippedPartitionByBroker = new HashMap<>();
     Map<Integer, Map<String, Integer>> leaderDistribution = leaderDistribution(cluster);
     for (TopicPartition tpDotNotHandled : partitionsDotNotHandled) {
       try {
-        PartitionMetricSample sample = buildPartitionMetricSample(cluster, leaderDistribution, tpDotNotHandled,
-                                                                  _brokerLoad, _maxMetricTimestamp, _cachedNumCoresByBroker);
+        PartitionMetricSample sample = buildPartitionMetricSample(cluster, leaderDistribution, tpDotNotHandled, _brokerLoad,
+                                                                  _maxMetricTimestamp, _cachedNumCoresByBroker, skippedPartitionByBroker);
         if (sample != null) {
           LOG.trace("Added partition metrics sample for {}.", tpDotNotHandled);
           partitionMetricSamples.add(sample);
-        } else {
-          skippedPartition++;
         }
       } catch (Exception e) {
         LOG.error("Error building partition metric sample for {}.", tpDotNotHandled, e);
-        skippedPartition++;
+        skippedPartitionByBroker.merge(-1, 1, Integer::sum);
```

Comment:
Nit: `-1` -> to a static var `UNRECOGNIZED_BROKER_ID` and let's update the return JavaDoc